internal/grpc/services/gateway/storageprovider.go

```diff
@@ -1248,147 +567,7 @@ func (s *svc) Stat(ctx context.Context, req *provider.StatRequest) (*provider.St
-		// So stat it now
-		if res == nil {
-			res, err = s.stat(ctx, req)
-			if err != nil {
-				return &provider.StatResponse{
-					Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+req.Ref.String()),
-				}, nil
-			}
-
-			if res.Status.Code != rpc.Code_CODE_OK {
-				return &provider.StatResponse{
-					Status: res.Status,
-				}, nil
-			}
-		}
-
-		ri, protocol, err := s.checkRef(ctx, res.Info)
-		if err != nil {
-			return &provider.StatResponse{
-				Status: status.NewStatusFromErrType(ctx, "error resolving reference "+res.Info.Target, err),
-			}, nil
-		}
-
-		if protocol == "webdav" {
-			ri, err = s.webdavRefStat(ctx, res.Info.Target)
-			if err != nil {
-				return &provider.StatResponse{
-					Status: status.NewInternal(ctx, err, "gateway: error resolving webdav reference: "+p),
-				}, nil
-			}
-		}
-
-		// we need to make sure we don't expose the reference target in the resource
-		// information. For example, if requests comes to: /home/MyShares/photos and photos
-		// is reference to /user/peter/Holidays/photos, we need to still return to the user
-		// /home/MyShares/photos
-		orgPath := res.Info.Path
-		res.Info = ri
-		res.Info.Path = orgPath
-		return res, nil
-	}
-
-	if s.isShareChild(ctx, p) {
-		shareName, shareChild := s.splitShare(ctx, p)
-
-		statReq := &provider.StatRequest{Ref: &provider.Reference{Path: shareName}}
-		statRes, err := s.stat(ctx, statReq)
-		if err != nil {
-			return &provider.StatResponse{
-				Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+statReq.Ref.String()),
-			}, nil
-		}
-
-		if statRes.Status.Code != rpc.Code_CODE_OK {
-			return &provider.StatResponse{
-				Status: statRes.Status,
-			}, nil
-		}
-
-		ri, protocol, err := s.checkRef(ctx, statRes.Info)
-		if err != nil {
-			return &provider.StatResponse{
-				Status: status.NewStatusFromErrType(ctx, "error resolving reference "+statRes.Info.Target, err),
-			}, nil
-		}
-
-		if protocol == "webdav" {
-			ri, err = s.webdavRefStat(ctx, statRes.Info.Target, shareChild)
-			if err != nil {
-				return &provider.StatResponse{
-					Status: status.NewInternal(ctx, err, "gateway: error resolving webdav reference: "+p),
-				}, nil
-			}
-			ri.Path = p
-			return &provider.StatResponse{
-				Status: status.NewOK(ctx),
-				Info:   ri,
-			}, nil
-		}
-
-		// append child to target
-		req.Ref.Path = path.Join(ri.Path, shareChild)
-		res, err := s.stat(ctx, req)
-		if err != nil {
-			return &provider.StatResponse{
-				Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+req.Ref.String()),
-			}, nil
-		}
-		if res.Status.Code != rpc.Code_CODE_OK {
-			return &provider.StatResponse{
-				Status: res.Status,
-			}, nil
-		}
-
-		// we need to make sure we don't expose the reference target in the resource
-		// information.
-		res.Info.Path = p
-		return res, nil
-	}
-
-	panic("gateway: stating an unknown path:" + p)
+	return s.stat(ctx, req)
```

Comment:
Same comment here:
a) Why the check for relative reference if we don't use it?
b) it's just a call to `stat`, so why even split into `Stat` and `stat`?